daisy/workflow.go

```diff
@@ -670,6 +670,13 @@ func (w *Workflow) runStep(ctx context.Context, s *Step) DError {
 	select {
 	case err := <-e:
 		return err
+	case <-ctx.Done():
```

Comment:
it seems that my terminal is dead when I press ctrl+c during running daisy cli today, is this PR fixing that?

Reply:
No. This is to support using context expiration for timeout handling, related to #1263